### ISSUE TITLE
Ctrl + BackSpace caused an exception

### ISSUE THREAD

**Issue by morphmetrix** [PowerShell/PSReadLine#462]
Hi Team. What an awesome mode PSReadLine is. I hit a speed bump though. See the output below.

-----------------------------------------------------------------------
Last 200 Keys:
 LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow
LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow Backspace Backspace [ ] LeftArrow ^ . Enter
 UpArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow Le
ftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow Lef
tArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow Left
Arrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftA
rrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftArrow LeftAr
row LeftArrow LeftArrow LeftArrow LeftArrow LeftWindows LeftWindows LeftWindows Home LeftArrow RightArrow End Ctrl+Delete Ctrl+Backspace

Exception:
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument argument, ExceptionResource resource)
   at Microsoft.PowerShell.PSConsoleReadLine.Kill(Int32 start, Int32 length, Boolean prepend)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------

**Comment by lzybkr** [comment]
Closing as a dupe of #531 - there's still not enough information here for me to reproduce. I've reviewed the code as well and I still can't construct a case where this happens. I believe there is a bug, so if you do see this stack again, maybe you can spend some time to try and reproduce or at least share a crash dump with me.

**Comment by kevin-he-01** [comment]
@lzybkr This can actually be easily reproduced by using a **clean** console (A newly launched one)
As long as the cursor position is **always** after the rightmost character, enter _any_ text **in the first line in the console**, then followed by `Ctrl + Delete` and finally `Ctrl + Backspace` **always** reproduce this bug.
```
Last 3 Keys:
 c Ctrl+Delete Ctrl+Backspace

Exception:
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.ThrowHelper.ThrowArgumentOutOfRange_IndexException()
   at System.Collections.Generic.List`1.get_Item(Int32 index)
   at Microsoft.PowerShell.PSConsoleReadLine.Kill(Int32 start, Int32 length, Boolean prepend)
   at Microsoft.PowerShell.PSConsoleReadLine.BackwardKillWord(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
```
The keys entered before the `Ctrl + Delete` + `Ctrl + Backspace` key sequence ("c" in this case) does not matter. The culprit of this bug seems to be the final  `Ctrl + Delete` + `Ctrl + Backspace` (`Ctrl + Delete` has no effect on the text as there's no characters after the cursor. Nevertheless, it cannot be omitted)

**PSReadLine version:** 1.2

**Comment by kevin-he-01** [comment]
@lzybkr Seems to be fixed in PSReadLine 2.0.0, which is the default for the latest version ([6.1.0](https://github.com/PowerShell/PowerShell/releases/tag/v6.1.0)) of [PowerShell Core](https://github.com/PowerShell/PowerShell): I can no longer reproduce it. To make it take effect in Windows PowerShell, run `powershell.exe` from PowerShell Core 6.1+ so it will inherit PSReadline 2.0.0 from it rather than the default version 1.2.

**Comment by daxian-dbw** [comment]
Closed as fixed.
